File: android/core/src/main/java/ru/maksimbulva/chess/core/engine/position/history/PositionHistoryItem.kt
package ru.maksimbulva.chess.core.engine.position.history

import ru.maksimbulva.chess.core.engine.move.Move
import ru.maksimbulva.chess.core.engine.position.CastlingAvailability

class PositionHistoryItem(
    val movePlayed: Move,
    val whiteCastlingAvailability: CastlingAvailability,
    val blackCastlingAvailability: CastlingAvailability,
    val enPassantCaptureColumn: Int?,
    val halfMoveClock: Int,
    val fullMoveNumber: Int
)

File: ChessApp/ChessEngine/Position/PositionHistoryItem.cs
using Optional;

namespace ChessEngine.Position
{
    internal sealed class PositionHistoryItem
    {
        public Move.Move Move { get; }
        public Option<int> EnPassantCaptureColumn { get; }

        public PositionHistoryItem(
            Move.Move move,
            Option<int> enPassantCaptureColumn)
        {
            Move = move;
            EnPassantCaptureColumn = enPassantCaptureColumn;
        }
    }
}
